modules/javafx.base/src/main/java/com/sun/javafx/collections/ObservableSequentialListWrapper.java

```diff
@@ -175,6 +175,15 @@ public E get(int index) {
 
     @Override
     public boolean addAll(int index, Collection<? extends E> c) {
+        if (index < 0 || index > size()) {
```

Comment:
In order to use `Objects.checkIndex` here, I would have to call it like this: `Objects.checkIndex(index, size() + 1)`.
`size() + 1` is necessary because it is valid to call this method with an `index` equal to the size of the list.

However, the exception message that would be produced for an out-of-bounds index would be incorrect, for example when adding elements at index 20 to a list that already contains 10 elements: "Index 20 out of bounds for length 11". Obviously, a list of 10 elements doesn't have a length of 11.